src/util/haptics.ts

```diff
@@ -0,0 +1,18 @@
+import { forwardHaptic, HapticType } from "../data/haptics";
+
+const hapticPatterns = {
+  success: [50, 50, 50],
+  warning: [100, 50, 100],
+  failure: [200, 100, 200],
+  light: [100],
+  medium: [200],
+  heavy: [300],
+  selection: [50],
+};
+
+export const haptic = (hapticType: HapticType) => {
```

Comment:
Let's not conflate concerns of requesting a vibration and vibrating.

Instead, we should just listen for haptic events on window and vibrate accordingly. Leave all places with forwardHaptic in there. That way things are a little more reusable.

Reply:
What is the right place to add the event listener?